AngularApp/src/app/diagnostic-data/components/insights/insights.component.html

```diff
@@ -5,7 +5,7 @@
   <div *ngFor="let insight of insights" class="insight-container">
     <div class="panel panel-default">
       <div class="panel-heading">
-        <h5 class="panel-title" [class.clickable-header]="insight.hasData()" (click)="insight.showDetails = insight.hasData() && !insight.showDetails">
+        <h5 class="panel-title" [class.clickable-header]="insight.hasData()" (click)="insight.showDetails = insight.hasData() && !insight.showDetails;logInsightClickEvent(insight.title, insight.showDetails);">
```

Comment:
if you are going to do multiple actions on click, I find it is easier to debug if you move this logic to the component inside a method then just call that method from the (click) action.